### PR TITLE
Adding --machine-id option.

### PR DESCRIPTION
**Pull request by gsegatti**
Allows users to provide their own Machine ID via CLI or configuration file.

### CODE REVIEW

File: mkosi.md [hunk 1]
```diff
@@ -653,6 +653,11 @@ a boolean argument: either "1", "yes", or "true" to enable, or "0",
   in building or further container import stages.  This option strips
   SELinux context attributes from the resulting tar archive.
 
+`MachineID=`, `--machine-id`
+
+: Set the machine's ID to the specified value. If unused, a random ID will
+be generated using uuid4().
```
Comment (DaanDeMeyer) [inline mkosi.md:659]: No need to detail implementation details here. Just mention that a random machine ID will be used while building the image and that the final image will ship without a machine ID.

File: mkosi/__init__.py [hunk 2]
```diff
@@ -6505,7 +6513,15 @@ def load_args(args: argparse.Namespace) -> MkosiArgs:
...
     # always the same
-    args.machine_id = uuid.uuid4().hex
+    if args.machine_id is None:
+        args.machine_id = uuid.uuid4().hex
+        args.machine_id_is_fixed = False
+    else:
+        args.machine_id_is_fixed = True
+        try:
+            uuid.UUID(hex=args.machine_id)
+        except:
```
Comment (behrmann) [inline mkosi/__init__.py:6523]:
```suggestion
        except ValueError:
```
Please always be explicit about what you are catching. :)

File: mkosi/__init__.py [hunk 3]
```diff
@@ -5600,6 +5607,7 @@ def create_parser() -> ArgumentParserMkosi:
         help="Turn on debugging output",
         choices=("run", "build-script", "workspace-command", "disk"),
     )
+
```
Comment (behrmann) [inline mkosi/__init__.py:5610]: Unrelated whitespace change.

File: mkosi/__init__.py (hunk 2 above)
Comment (DaanDeMeyer) [inline mkosi/__init__.py:6523]: This should be `except Exception`

**Review by behrmann — COMMENTED** [review]
LGTM.

File: mkosi/__init__.py (hunk 2 above)
Comment (behrmann) [reply mkosi/__init__.py:6523]: Ah, you were a bit quicker than me. :) I think something more specific than `Exception` would better, though.

Comment (gsegatti) [reply mkosi/__init__.py]: You're absolutely right, thanks for the suggestion!

File: mkosi.md [hunk 4]
```diff
@@ -653,6 +653,12 @@ a boolean argument: either "1", "yes", or "true" to enable, or "0",
   in building or further container import stages.  This option strips
   SELinux context attributes from the resulting tar archive.
 
+`MachineID=`, `--machine-id`
+
+: Set the machine's ID to the specified value. If unused, a random ID will
+be used while building the image. The final image will be shipped without
```
Comment (DaanDeMeyer) [inline mkosi.md:659]:
```suggestion
be used while building the image and the final image will be shipped without
```

**Review by anitazha — COMMENTED** [review]
Did you mean to squash the two commits? Usually we don't have separate commits for "historical" changes (e.g. minor updates to a former commit in the same PR)
